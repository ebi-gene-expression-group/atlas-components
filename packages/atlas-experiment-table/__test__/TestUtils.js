import React from 'react'
import styled from 'styled-components'

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const TableCellDiv = styled.div`
  font-size: 13px;
  font-family: Helvetica, Arial, FreeSans, "Liberation Sans", sans-serif;
`

const tableHeader = [
  { type: `sort`, title: `Loaded date`, width: 140, dataParam: `lastUpdate` },
  { type: `search`, title: `species`, width: 200, dataParam: `species` },
  { type: ``, title: `experiment description`, width: 360, dataParam: `experimentDescription`, link: `experimentAccession`, resource: `experiments`, endpoint: `Results` },
  { type: `plain`, title: `experiment factors`, width: 260, dataParam: `experimentalFactors` },
  { type: `sort`, title: `Number of assays`, width: 160, dataParam: `numberOfAssays`, link: `experimentAccession`, resource: `experiments`, endpoint: `Experiment Design` },
]

const data = [
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-EHCA-2`,
    experimentDescription: `Melanoma infiltration`,
    lastUpdate: `16-11-2018`,
    numberOfAssays: 6638,
    numberOfContrasts: 0,
    species: `Mus musculus`,
    kingdom: `animals`,
    experimentalFactors: [`single cell identifier`, `sampling site`,`time`],
  },
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-GEOD-99058`,
    experimentDescription: `Single cell`,
    lastUpdate: `11-10-2018`,
    numberOfAssays: 250,
    numberOfContrasts: 0,
    species: `Mus musculus`,
    kingdom: `animals`,
    experimentalFactors: [`single cell identifier`]
  },
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-MTAB-5061`,
    experimentDescription: `healthy individuals and type 2 diabetes patients`,
    lastUpdate: `11-10-2018`,
    numberOfAssays: 3514,
    numberOfContrasts: 0,
    species: `Homo sapiens`,
    kingdom: `plants`,
    experimentalFactors: [`single cell identifier`, `disease`],
  }
]

export {getRandomInt, TableCellDiv, tableHeader, data}
