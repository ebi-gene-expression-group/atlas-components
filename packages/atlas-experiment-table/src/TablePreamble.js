import React from 'react'
import PropTypes from 'prop-types'

const TablePreamble =
({
  dropdowns,
  dropdownOnChange,
  rowsCount,
  rowsPerPageOptions,
  rowsPerPage,
  rowsPerPageOnChange,
  searchAll,
  searchAllOnChange
}) =>
  <div className={`row expanded`}>
    {
      dropdowns.map((dropdown, index) =>
        <div
          key={index}
          className={`small-12 medium-4 large-2 columns`}>
          <label>{dropdown.label}:
            <select
              defaultValue={dropdown.value || ``}
              onChange={e => dropdownOnChange(dropdown.dataKey, e.target.value)}>
              <option value={``}>All</option>
              {
                dropdown.options
                  .concat(dropdown.value && !dropdown.options.includes(dropdown.value) ? dropdown.value : [])
                  .map((option, index) => <option key={index} value={option}>{option}</option>)
              }
            </select>
          </label>
        </div>)
    }

    <div className={`small-12 medium-4 large-2 columns`}>
      <label>Entries per page:
        <select
          defaultValue={rowsPerPage}
          onChange={e => rowsPerPageOnChange(Number.parseInt(e.target.value))}>
          {
            rowsPerPageOptions
              .filter(rowsPerPageOption => rowsCount >= rowsPerPageOption)
              .map((rowsPerPageOption, index) =>
                <option key={index} value={rowsPerPageOption}>{rowsPerPageOption}</option>)
          }
          <option value={0}>All</option>
        </select>
      </label>
    </div>

    <div className={`small-12 medium-4 large-2 columns`}>
      <label>Search all columns:
        <input
          type={`search`}
          value={searchAll}
          onChange={e => searchAllOnChange(e.target.value)}/>
      </label>
    </div>
  </div>

TablePreamble.propTypes = {
  dropdowns: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    dataKey: PropTypes.string.isRequired,
    value: PropTypes.string,
    options: PropTypes.arrayOf(PropTypes.string).isRequired
  })).isRequired,
  dropdownOnChange: PropTypes.func.isRequired,
  rowsCount: PropTypes.number.isRequired,
  rowsPerPageOptions: PropTypes.arrayOf(PropTypes.number).isRequired,
  rowsPerPage: PropTypes.number.isRequired,
  rowsPerPageOnChange: PropTypes.func.isRequired,
  searchAll: PropTypes.string.isRequired,
  searchAllOnChange: PropTypes.func.isRequired
}

export default TablePreamble
