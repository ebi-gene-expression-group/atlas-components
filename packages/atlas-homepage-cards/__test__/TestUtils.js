

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const dummyCards = () => {
  return [
    {
      iconType: `species`,
      iconSrc: `mus musculus`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
    {
      iconType: `species`,
      iconSrc: `homo sapiens`,
      iconDescription: `Homo sapiens`,
      content: [
        {
          text: `2 experiments`,
        }
      ]
    },
    {
      iconType: `unknown`,
      iconSrc: `http://www.lingyun.com`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
    {
      iconType: `experiments`,
      iconSrc: `http://www.zhao.com`,
      iconDescription: `Homo sapiens`,
      content: [
        {
          text: `2 experiments`,
        }
      ]
    },
    {
      iconType: `image`,
      iconSrc: `logo-goes-here`,
      iconDescription: `An image card`,
      content: [
        {
          text: `Go to experiments`,
        }
      ]
    },
    {
      iconType: `imagespecies`,
      iconSrc: `http://www.lingyun.com`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
  ]
}

export {getRandomInt, dummyCards}