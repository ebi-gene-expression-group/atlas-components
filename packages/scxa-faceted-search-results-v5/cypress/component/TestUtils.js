import facets from '../fixtures/vindicatorsResponse.json'
import organismParts from '../fixtures/organismPartsResponse.json'
import cellTypes from '../fixtures/cellTypesResponse.json'
import species from '../fixtures/speciesResponse.json'

import PropTypes from "prop-types";
import styled from "styled-components";
import React from "react";
import CheckboxFacetGroupsDefaultProps from "../../src/facetgroups/CheckboxFacetGroupsDefaultProps";
import MultiSelectDropdownFacetGroupsDefaultProps
  from "../../src/facetgroups/MultiSelectDropdownFacetGroupsDefaultProps";

const getPropsForCheckBoxGroupWithTooltip = () => {
  return Cypress._.shuffle(CheckboxFacetGroupsDefaultProps).pop()
}

const getPropsForMultiSelectDropdownGroupWithTooltip = () => {
  return Cypress._.shuffle(MultiSelectDropdownFacetGroupsDefaultProps).pop()
}

const getPropsForMarkerGeneFacet = () => {
  const markerGeneName = `Marker genes`;
  return CheckboxFacetGroupsDefaultProps.find(facet => facet.name === markerGeneName)
}

const getPropsWithoutTooltip = () => {
  let props = getPropsForCheckBoxGroupWithTooltip()
  props.description = ``
  return props
}

const getFacets = () => {
  return Cypress._.shuffle(facets).filter(() => Math.random() > 0.5)
}

const getRandomSpecies = () => {
  return Cypress._.shuffle(species).filter(() => Math.random() > 0.5)
}

const getRandomOrganismParts = () => {
  return Cypress._.shuffle(organismParts).filter(() => Math.random() > 0.5)
}

const getRandomCellTypes = () => {
  return Cypress._.shuffle(cellTypes).filter(() => Math.random() > 0.5)
}

const ExperimentTableCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

ExperimentTableCard.propTypes = {
  title: PropTypes.string.isRequired
}

const IconDiv = styled.div`
  width: 15%;
  text-align: center;
`
IconDiv.displayName = 'IconDiv'

const BitDiv = styled.div`
  width: 25%;
  text-align: center;
`
const DoveDiv = styled.div`
  width: 25%;
  text-align: center;
`

const ExperimentTableHeader = ({onClick}) =>
  <DoveDiv>
    <IconDiv onClick={()=>onClick(`test`)}>
      Meow
    </IconDiv>
    <BitDiv onClick={()=>onClick(`test`)}>
      Wow wow
    </BitDiv>
  </DoveDiv>

ExperimentTableHeader.propTypes = {
  onClick: PropTypes.func.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

export { getRandomInt, getFacets, getRandomOrganismParts, getRandomCellTypes, getRandomSpecies,
  getPropsForCheckBoxGroupWithTooltip, getPropsForMultiSelectDropdownGroupWithTooltip,
  getPropsWithoutTooltip, getPropsForMarkerGeneFacet,
  ExperimentTableCard, ExperimentTableHeader }
