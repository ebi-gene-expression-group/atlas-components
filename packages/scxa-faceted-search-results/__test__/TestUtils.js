import React from 'react'
import PropTypes from 'prop-types'

const EpisodeCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

EpisodeCard.propTypes = {
  title: PropTypes.string.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min)) + min; //The maximum is exclusive and the minimum is inclusive
}

const vindicators = [
  { group:`Vindicator`, value: `supernova`, label: `Supernova`},
  { group:`Vindicator`, value: `noob-noob`, label: `Noob-Noob` },
  { group:`Vindicator`, value: `vance_maximus`, label: `Vance Maximus`},
  { group:`Vindicator`, value: `alan_rails`, label: `Alan Rails`},
  { group:`Vindicator`, value: `crocubot`, label: `Crocubot` },
  { group:`Vindicator`, value: `million_ants`, label: `Million Ants`},
  { group:`Vindicator`, value: `morty_smith`, label: `Morty Smith`},
  { group:`Vindicator`, value: `rick_sanchez`, label: `Rick Sanchez`},
  { group:`Vindicator`, value: `lady_katana`, label: `Lady Katana`},
  { group:`Vindicator`, value: `calypso`, label: `Calypso`},
  { group:`Vindicator`, value: `diablo_verde`, label: `Diablo Verde`}
]

const episodes = [
  {
    element: {
      title: "Raising Gazorpazorp"
    },
    facets: [
      {
        group: "Planet",
        value: "gazorpazorp",
        label: "Gazorpazorp"
      },
      {
        group: "Guest character",
        value: "gwendolyn",
        label: "Gwendolyn"
      },
      {
        group: "Guest character",
        value: "ma-sha",
        label: "Ma-Sha"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  },
  {
    element: {
      title: "The wedding squanchers"
    },
    facets: [
      {
        group: "Planet",
        value: "squanch",
        label: "Squanch"
      },
      {
        group: "Guest character",
        value: "birdperson",
        label: "Birdperson"
      },
      {
        group: "Guest character",
        value: "squanchy",
        label: "Squanchy"
      },
      {
        group: "Season",
        value: "2",
        label: "2"
      }
    ]
  },
  {
    element: {
      title: "The Rickshank redemption"
    },
    facets: [
      {
        group: "Guest character",
        value: "birdperson",
        label: "Birdperson"
      },
      {
        group: "Planet",
        value: "buttworld",
        label: "Buttworld"
      },
      {
        group: "Season",
        value: "3",
        label: "3"
      }
    ]
  },
  {
    element: {
      title: "Ricksy business"
    },
    facets: [
      {
        group: "Guest character",
        value: "squanchy",
        label: "Squanchy"
      },
      {
        group: "Guest character",
        value: "abradolf_lincler",
        label: "Abradolf Lincler"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  },
  {
    element: {
      title: "Close Rick-counters of the Rick kind"
    },
    facets: [
      {
        group: "Guest character",
        value: "ricktiminus_sancheziminius",
        label: "Ricktiminus Sancheziminius"
      },
      {
        group: "Guest character",
        value: "abradolf_lincler",
        label: "Abradolf Lincler"
      },
      {
        group: "Planet",
        value: "buttworld",
        label: "Buttworld"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  }
]



export {getRandomInt, vindicators, episodes, EpisodeCard}
