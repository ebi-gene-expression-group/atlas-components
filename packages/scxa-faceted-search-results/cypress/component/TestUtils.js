import episodes from '../fixtures/episodes.json'
import PropTypes from "prop-types";
import styled from "styled-components";
import React from "react";

const facets = () => {
  return episodes.reduce((acc, episode) => acc.concat(episode.facets), [])
}

const getUniqueFacets = () => {
  const allFacets = facets()
  return allFacets
    .filter((facet, index) => allFacets.findIndex((thatFacet) => facet.value === thatFacet.value) === index)
    .map((facet) => ({
      ...facet,
      disabled: false
    }))
}

const getFacetTooltip = () => {
  return Cypress._.shuffle(getUniqueFacets()).find(facet => facet.description)
}

const getFacetWithoutTooltip = () => {
  return Cypress._.shuffle(getUniqueFacets()).find(facet => !facet.description)
}

const getPropsWithTooltip = (facetTooltip) => {
  return {
    facetGroupName: facetTooltip.group,
      facetGroupNameDescription: facetTooltip.description,
    facets: getUniqueFacets().filter(facet => facet.group === facetTooltip.group),
    onChange: () => {}
  }
}

const getPropsWithoutTooltip = () => {
  const facetWithoutTooltip = getFacetWithoutTooltip()
  return {
    facetGroupName: facetWithoutTooltip.group,
      facets: getUniqueFacets().filter(facet => facet.group === facetWithoutTooltip.group),
    onChange: () => {}
  }
}

const ExperimentTableCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

ExperimentTableCard.propTypes = {
  title: PropTypes.string.isRequired
}

const IconDiv = styled.div`
  width: 15%;
  text-align: center;
`
IconDiv.displayName = 'IconDiv'

const BitDiv = styled.div`
  width: 25%;
  text-align: center;
`
const DoveDiv = styled.div`
  width: 25%;
  text-align: center;
`

const ExperimentTableHeader = ({onClick}) =>
  <DoveDiv>
    <IconDiv onClick={()=>onClick(`test`)}>
      Meow
    </IconDiv>
    <BitDiv onClick={()=>onClick(`test`)}>
      Wow wow
    </BitDiv>
  </DoveDiv>

ExperimentTableHeader.propTypes = {
  onClick: PropTypes.func.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

export { episodes, getRandomInt, getUniqueFacets, getFacetTooltip, getPropsWithTooltip, getPropsWithoutTooltip,
  ExperimentTableCard, ExperimentTableHeader }
