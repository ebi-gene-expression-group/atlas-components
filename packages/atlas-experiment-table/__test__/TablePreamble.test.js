import React from 'react'
import { shallow, mount } from 'enzyme'

import TablePreamble from '../src/TablePreamble'

import randomString from 'random-string'
import { getRandomInt } from './TestUtils'

describe(`TablePreamble`, () => {
  const MAX_ROW_COUNT = 1000
  const MAX_DROPDOWN_COUNT = 4
  const MAX_OPTIONS_COUNT = 10

  const props = {
    dropdowns: [],
    dropdownOnChange: jest.fn(),
    rowsCount: getRandomInt(100, MAX_ROW_COUNT),
    rowsPerPageOptions: [10, 25, 50],
    rowsPerPage: 10,
    rowsPerPageOnChange: jest.fn(),
    searchAll: randomString(),
    searchAllOnChange: jest.fn()
  }

  test(`displays, at the very least, a page size dropdown and a global search box`, () => {
    const wrapper = shallow(<TablePreamble {...props}/>)

    expect(wrapper).toContainExactlyOneMatchingElement(`select`)
    expect(wrapper).toContainExactlyOneMatchingElement(`input`)
  })

  test(`can have an initial value passed to the global search box`, () => {
    const wrapper = shallow(<TablePreamble {...props}/>)

    expect(wrapper.find(`input`)).toHaveProp(`value`, props.searchAll)
  })

  test(`calls searchAllOnChange when the global text box changes`, () => {
    const wrapper = shallow(<TablePreamble {...props}/>)

    const event = { target: { value: randomString() } }
    wrapper.find(`input`).simulate(`change`, event)

    expect(props.searchAllOnChange).toHaveBeenCalled()
    expect(props.searchAllOnChange.mock.calls).toContainEqual([event.target.value])
  })

  test(`adds an ‘All’ option to the page size dropdown if all options are smaller than the number of rows`, () => {
    const wrapper = shallow(<TablePreamble {...props}/>)

    expect(wrapper.find(`option`)).toHaveLength(props.rowsPerPageOptions.length + 1)
    expect(wrapper.find(`option`).last()).toHaveText(`All`)
    expect(wrapper.find(`option`).last()).toHaveProp(`value`, 0)
  })

  test(`calls rowsPerPageOnChange when the page size dropdown is changed`, () => {
    const wrapper = shallow(<TablePreamble {...props}/>)

    const event = { target: { value: props.rowsPerPageOptions[getRandomInt(0, props.rowsPerPageOptions.length)] } }
    wrapper.find(`select`).simulate(`change`, event)

    expect(props.rowsPerPageOnChange).toHaveBeenCalled()
    expect(props.rowsPerPageOnChange.mock.calls).toContainEqual([event.target.value])
  })

  test(`renders as many dropdowns as specified in the dropdowns prop`, () => {
    const dropdowns =
      // https://stackoverflow.com/questions/3751520/how-to-generate-sequence-of-numbers-chars-in-javascript
      Array.apply(0, Array(getRandomInt(1, MAX_DROPDOWN_COUNT)))
        .map(() => (
          {
            label: randomString(),
            dataKey: randomString(),
            options: []
          }
        ))

    const wrapper = shallow(<TablePreamble {...props} dropdowns={dropdowns}/>)

    expect(wrapper.find(`select`)).toHaveLength(dropdowns.length + 1)
  })

  test(`adds an ‘All’ option with an empty value to dropdowns`, () => {
    const options =
      // https://stackoverflow.com/questions/3751520/how-to-generate-sequence-of-numbers-chars-in-javascript
      Array.apply(0, Array(getRandomInt(1, MAX_OPTIONS_COUNT)))
        .map(() => randomString())

    const dropdowns = [
      {
        label: randomString(),
        dataKey: randomString(),
        options: options
      }
    ]

    const wrapper = shallow(<TablePreamble {...props} dropdowns={dropdowns}/>)

    expect(wrapper.find(`select`).first().find(`option`)).toHaveLength(options.length + 1)
    expect(wrapper.find(`select`).first().find(`option`).first()).toHaveText(`All`)
    expect(wrapper.find(`select`).first().find(`option`).first()).toHaveProp(`value`, ``)
  })

  test(`can fuzzily match the initial value in a dropdown filter if it isn’t an exact match of an option`, () => {
    const options =
      // https://stackoverflow.com/questions/3751520/how-to-generate-sequence-of-numbers-chars-in-javascript
      Array.apply(0, Array(getRandomInt(1, MAX_OPTIONS_COUNT)))
        .map(() => randomString().toUpperCase())

    const randomOptionIndex = getRandomInt(0, options.length)

    const dropdowns = [
      {
        label: randomString(),
        dataKey: randomString(),
        options: options,
        value: `  ` + options[randomOptionIndex].toLowerCase() + `\t   `
      }
    ]

    const wrapper = shallow(<TablePreamble {...props} dropdowns={dropdowns}/>)

    expect(wrapper.find(`select`).first().find(`option`)).toHaveLength(options.length + 1)
    expect(wrapper.find(`select`).first()).toHaveProp({value: options[randomOptionIndex]})
  })

  test(`defaults to ‘All’  if the initial value in a dropdown filter isn’t one of the options`, () => {
    const options =
      // https://stackoverflow.com/questions/3751520/how-to-generate-sequence-of-numbers-chars-in-javascript
      Array.apply(0, Array(getRandomInt(1, MAX_OPTIONS_COUNT)))
        .map(() => randomString())

    const dropdowns = [
      {
        label: randomString(),
        dataKey: randomString(),
        options: options,
        value: randomString()
      }
    ]

    const wrapper = shallow(<TablePreamble {...props} dropdowns={dropdowns}/>)

    expect(wrapper.find(`select`).first().find(`option`)).toHaveLength(options.length + 1)
    expect(wrapper.find(`select`).first()).toHaveProp({value: ``})
  })

  test(`renders as many dropdowns as specified in the dropdowns prop`, () => {
    const dropdowns =
      // https://stackoverflow.com/questions/3751520/how-to-generate-sequence-of-numbers-chars-in-javascript
      Array.apply(0, Array(getRandomInt(1, MAX_DROPDOWN_COUNT)))
        .map(() => (
          {
            label: randomString(),
            dataKey: randomString(),
            options: []
          }
        ))

    const wrapper = shallow(<TablePreamble {...props} dropdowns={dropdowns}/>)

    expect(wrapper.find(`select`)).toHaveLength(dropdowns.length + 1)
  })

  test(`calls dropdownOnChange with an explicit flag not to debounce filtering`, () => {
    const dropdowns =
      // https://stackoverflow.com/questions/3751520/how-to-generate-sequence-of-numbers-chars-in-javascript
      Array.apply(0, Array(getRandomInt(1, MAX_DROPDOWN_COUNT)))
        .map(() => (
          {
            label: randomString(),
            dataKey: randomString(),
            options:
              Array.apply(0, Array(getRandomInt(1, MAX_OPTIONS_COUNT)))
                .map(() => randomString())
          }
        ))

    const wrapper = shallow(<TablePreamble {...props} dropdowns={dropdowns}/>)

    const event = { target: { value: randomString() } }
    const randomDropdownIndex = getRandomInt(0, dropdowns.length)
    wrapper.find(`select`).at(randomDropdownIndex).simulate(`change`, event)

    expect(props.dropdownOnChange).toHaveBeenCalled()
    expect(props.dropdownOnChange.mock.calls)
      .toContainEqual([dropdowns[randomDropdownIndex].dataKey, event.target.value, false])
  })

  test(`matches snapshot (basic)`, () => {
    const props = {
      dropdowns: [],
      dropdownOnChange: jest.fn(),
      rowsCount: 1000,
      rowsPerPageOptions: [10, 25, 50],
      rowsPerPage: 10,
      rowsPerPageOnChange: jest.fn(),
      searchAll: ``,
      searchAllOnChange: jest.fn()
    }

    expect(mount(<TablePreamble {...props}/>)).toMatchSnapshot()
  })

  test(`matches snapshot (with dropdowns)`, () => {
    const props = {
      dropdowns: [
        {
          label: `Experiment type`,
          dataKey: `experimentType`,
          options: [`Baseline`, `Differential`],
        },
        {
          label: `Kingdom`,
          dataKey: `kingdom`,
          options: [`Animals`, `Plants`, `Fungi`],
          value: `Dorne`
        },
        {
          label: `Technology type`,
          dataKey: `technologyType`,
          options: [`Smart-Seq`, `10x`, `Droplet-based`],
          value: `Smart-Seq`
        }
      ],
      dropdownOnChange: jest.fn(),
      rowsCount: 1000,
      rowsPerPageOptions: [10, 25, 50],
      rowsPerPage: 10,
      rowsPerPageOnChange: jest.fn(),
      searchAll: ``,
      searchAllOnChange: jest.fn()
    }

    expect(mount(<TablePreamble {...props}/>)).toMatchSnapshot()
  })
})
