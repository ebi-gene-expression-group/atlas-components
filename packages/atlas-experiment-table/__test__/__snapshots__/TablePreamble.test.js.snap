// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TablePreamble matches snapshot (basic) 1`] = `
<TablePreamble
  dropdownOnChange={[MockFunction]}
  dropdowns={Array []}
  rowsCount={1000}
  rowsPerPage={10}
  rowsPerPageOnChange={[MockFunction]}
  rowsPerPageOptions={
    Array [
      10,
      25,
      50,
    ]
  }
  searchAll=""
  searchAllOnChange={[MockFunction]}
>
  <div
    className="row expanded"
  >
    <div
      className="small-12 medium-4 large-2 columns"
    >
      <label>
        Entries per page:
        <select
          defaultValue={10}
          onChange={[Function]}
        >
          <option
            key="0"
            value={10}
          >
            10
          </option>
          <option
            key="1"
            value={25}
          >
            25
          </option>
          <option
            key="2"
            value={50}
          >
            50
          </option>
          <option
            value={0}
          >
            All
          </option>
        </select>
      </label>
    </div>
    <div
      className="small-12 medium-4 large-2 columns"
    >
      <label>
        Search all columns:
        <input
          onChange={[Function]}
          type="search"
          value=""
        />
      </label>
    </div>
  </div>
</TablePreamble>
`;

exports[`TablePreamble matches snapshot (with dropdowns) 1`] = `
<TablePreamble
  dropdownOnChange={[MockFunction]}
  dropdowns={
    Array [
      Object {
        "dataKey": "experimentType",
        "label": "Experiment type",
        "options": Array [
          "Baseline",
          "Differential",
        ],
      },
      Object {
        "dataKey": "kingdom",
        "label": "Kingdom",
        "options": Array [
          "Animals",
          "Plants",
          "Fungi",
        ],
        "value": "Dorne",
      },
      Object {
        "dataKey": "technologyType",
        "label": "Technology type",
        "options": Array [
          "Smart-Seq",
          "10x",
          "Droplet-based",
        ],
        "value": "Smart-Seq",
      },
    ]
  }
  rowsCount={1000}
  rowsPerPage={10}
  rowsPerPageOnChange={[MockFunction]}
  rowsPerPageOptions={
    Array [
      10,
      25,
      50,
    ]
  }
  searchAll=""
  searchAllOnChange={[MockFunction]}
>
  <div
    className="row expanded"
  >
    <div
      className="small-12 medium-4 large-2 columns"
      key="0"
    >
      <label>
        Experiment type
        :
        <select
          defaultValue=""
          onChange={[Function]}
        >
          <option
            value=""
          >
            All
          </option>
          <option
            key="0"
            value="Baseline"
          >
            Baseline
          </option>
          <option
            key="1"
            value="Differential"
          >
            Differential
          </option>
        </select>
      </label>
    </div>
    <div
      className="small-12 medium-4 large-2 columns"
      key="1"
    >
      <label>
        Kingdom
        :
        <select
          defaultValue="Dorne"
          onChange={[Function]}
        >
          <option
            value=""
          >
            All
          </option>
          <option
            key="0"
            value="Animals"
          >
            Animals
          </option>
          <option
            key="1"
            value="Plants"
          >
            Plants
          </option>
          <option
            key="2"
            value="Fungi"
          >
            Fungi
          </option>
          <option
            key="3"
            value="Dorne"
          >
            Dorne
          </option>
        </select>
      </label>
    </div>
    <div
      className="small-12 medium-4 large-2 columns"
      key="2"
    >
      <label>
        Technology type
        :
        <select
          defaultValue="Smart-Seq"
          onChange={[Function]}
        >
          <option
            value=""
          >
            All
          </option>
          <option
            key="0"
            value="Smart-Seq"
          >
            Smart-Seq
          </option>
          <option
            key="1"
            value="10x"
          >
            10x
          </option>
          <option
            key="2"
            value="Droplet-based"
          >
            Droplet-based
          </option>
        </select>
      </label>
    </div>
    <div
      className="small-12 medium-4 large-2 columns"
    >
      <label>
        Entries per page:
        <select
          defaultValue={10}
          onChange={[Function]}
        >
          <option
            key="0"
            value={10}
          >
            10
          </option>
          <option
            key="1"
            value={25}
          >
            25
          </option>
          <option
            key="2"
            value={50}
          >
            50
          </option>
          <option
            value={0}
          >
            All
          </option>
        </select>
      </label>
    </div>
    <div
      className="small-12 medium-4 large-2 columns"
    >
      <label>
        Search all columns:
        <input
          onChange={[Function]}
          type="search"
          value=""
        />
      </label>
    </div>
  </div>
</TablePreamble>
`;
